packages/@n8n/nodes-langchain/nodes/vector_store/VectorStoreMongoDBAtlas/VectorStoreMongoDBAtlas.node.ts

```diff
@@ -0,0 +1,180 @@
+	},
+	methods: { listSearch: { mongoCollectionSearch } },
+	retrieveFields,
+	loadFields: retrieveFields,
+	insertFields,
+	sharedFields,
+	async getVectorStoreClient(context, filter, embeddings, itemIndex) {
+		try {
+			const collectionName = context.getNodeParameter('mongoCollection', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const vectorIndexName = context.getNodeParameter('vectorIndexName', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const embeddingField = context.getNodeParameter('embedding', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const metadataField = context.getNodeParameter('metadata_field', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const credentials = await context.getCredentials('mongoDb');
+			const client = new MongoClient(credentials.connectionString as string, {
+				appName: 'devrel.content.n8n_vector_integ', // Added appName
+			});
+			await client.connect();
+			const collection = client.db(credentials.database as string).collection(collectionName);
+
+			// test index exists
+			const indexes = await collection.listSearchIndexes().toArray();
+
+			const indexExists = indexes.some((index) => index.name === vectorIndexName);
+
+			if (!indexExists) {
+				throw new NodeOperationError(context.getNode(), `Index ${vectorIndexName} not found`, {
+					itemIndex,
+					description: 'Please check that the index exists in your collection',
+				});
+			}
+
+			return new MongoDBAtlasVectorSearch(embeddings, {
+				collection,
+				indexName: vectorIndexName, // Default index name
+				textKey: metadataField, // Field containing raw text
+				embeddingKey: embeddingField, // Field containing embeddings
+			});
+		} catch (error) {
+			throw new NodeOperationError(context.getNode(), `Error: ${error.message}`, {
+				itemIndex,
+				description: 'Please check your MongoDB Atlas connection details',
+			});
+		}
+	},
+	async populateVectorStore(context, embeddings, documents, itemIndex) {
+		try {
+			const collectionName = context.getNodeParameter('mongoCollection', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+			const embeddingField = context.getNodeParameter('embedding', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const metadataField = context.getNodeParameter('metadata_field', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const vectorIndexName = context.getNodeParameter('vectorIndexName', itemIndex, '', {
+				extractValue: true,
+			}) as string;
+
+			const credentials = await context.getCredentials('mongoDb');
+
+			const client = new MongoClient(credentials.connectionString as string, {
+				appName: 'devrel.content.n8n_vector_integ', // Added appName
+			});
+			await client.connect();
+
+			const db = client.db(credentials.database as string);
+
+			// Check if collection exists
+			const collections = await db.listCollections({ name: collectionName }).toArray();
```

Comment:
This bit of code is repeated 3 times. maybe we can extract this out into a shared function.

Reply:
I've submitted the fix for this.